src/core/openthread.cpp

```diff
@@ -977,33 +979,33 @@ void HandleActiveScanResult(void *aContext, Mac::Frame *aFrame)
         memcpy(&result.mExtendedPanId, beacon->GetExtendedPanId(), sizeof(result.mExtendedPanId));
     }
 
-    handler(&result);
+    handler(&result, aClientContext);
 
 exit:
     return;
 }
 
 ThreadError otDiscover(uint32_t aScanChannels, uint16_t aScanDuration, uint16_t aPanId,
-                       otHandleActiveScanResult aCallback)
+                       otHandleActiveScanResult aCallback, void *aCallbackContext)
 {
     return sThreadNetif->GetMle().Discover(aScanChannels, aScanDuration, aPanId, &HandleMleDiscover,
-                                           reinterpret_cast<void *>(aCallback));
+                                           reinterpret_cast<void *>(aCallback), aCallbackContext);
```

Comment:
I think it still makes sense for `Mle::Discover()` to take a context pointer since this API may be used internally by other components in OT.  I'm thinking about when we fully hook in Thread Commissioning, the Joiner component may need to perform a discover before initiating the DTLS handshake.